src/python/pants/backend/scala/compile/scalac_plugins.py

```diff
@@ -3,41 +3,152 @@
 
 from __future__ import annotations
 
-from typing import cast
+from dataclasses import dataclass
+from typing import Iterator, cast
 
-from pants.jvm.resolve.jvm_tool import JvmToolBase
-from pants.util.docutil import git_url
+from pants.backend.scala.target_types import (
+    ScalacPluginArtifactField,
+    ScalacPluginNameField,
+    ScalacPluginTarget,
+)
+from pants.build_graph.address import AddressInput
+from pants.engine.rules import Get, MultiGet, collect_rules, rule
+from pants.engine.target import WrappedTarget
+from pants.engine.unions import UnionRule
+from pants.jvm.resolve.coursier_fetch import (
+    CoursierResolvedLockfile,
+    MaterializedClasspath,
+    MaterializedClasspathRequest,
+)
+from pants.jvm.resolve.jvm_tool import JvmToolLockfileRequest, JvmToolLockfileSentinel
+from pants.jvm.resolve.jvm_tool import rules as jvm_tool_rules
+from pants.option.subsystem import Subsystem
+from pants.util.ordered_set import FrozenOrderedSet
+from pants.util.strutil import bullet_list
 
 
-class ScalacPlugins(JvmToolBase):
+class ScalacPlugins(Subsystem):
     options_scope = "scalac-plugins"
-    help = (
-        "Plugins for `scalac`.\n\n"
-        "Each artifact specified in `--artifacts` should be loadable as a `scalac` plugin. "
-        "Note that the `--version` flag has no meaning in this case, and needn't be set."
-    )
-
-    default_artifacts = ()
-    default_lockfile_resource = (
-        "pants.backend.scala.compile",
-        "scalac_plugins.default.lockfile.txt",
-    )
-    default_lockfile_url = git_url(
-        "src/python/pants/backend/scala/compile/scalac_plugins.default.lockfile.txt"
-    )
+    help = "Plugins for `scalac`."
 
     @classmethod
     def register_options(cls, register):
         super().register_options(register)
         register(
-            "--names",
+            "--global",
             type=list,
             member_type=str,
             advanced=True,
             default=[],
-            help=("The list of plugin names for the associated `--artifacts`."),
+            # NB: `global` is a python keyword.
+            dest="global_addresses",
+            help=(
+                "A list of addresses of `scalac_plugin` targets which should be used for "
+                "compilation of all Scala targets in a build."
+            ),
         )
+        register(
+            "--lockfile",
+            type=str,
+            default="3rdparty/jvm/global_scalac_plugins.lockfile",
```

Comment:
Does need a TODO comment to flag it for whatever Eric comes up with in reworking lockfiles?

Reply:
Never mind, there is no default resource, so my comment is moot.